hail/python/hail/typecheck2/check.py

```diff
@@ -282,7 +282,16 @@ def check_t(arg, t, context: str) -> None:
     List: check_list,
     Tuple: check_tuple,
     Collection: check_collection,
-    Callable: check_callable
+    Callable: check_callable,
```

Comment:
this is fine. I'll make a PR after yours to just delete `typecheck2` - it was an experiment that never made it to live use, and I now think there are other ways to achieve the benefit it would have provided.